include/wx/debug.h

```diff
@@ -390,6 +390,32 @@ extern void WXDLLIMPEXP_BASE wxAbort();
 //     to begin with...)
 #define wxCHECK_RET(cond, msg)       wxCHECK2_MSG(cond, return, msg)
 
+#if wxDEBUG_LEVEL
+
+// This macro checks if the evaluation of cond, having a return value of
+// OS Error type, is zero, ie no error occurred, and calls the assert handler
+// with the provided message if it isn't and finally traps if the special flag
+// indicating that it should do it was set by the handler.
+#define wxVERIFY_NOERR(cond)                \
+    wxSTATEMENT_MACRO_BEGIN                                               \
+        unsigned long evalOnceErrorCode = (cond);                         \
+        if ( evalOnceErrorCode == 0 )                                     \
+        {                                                                 \
+        }                                                                 \
+        else                                                              \
+        {                                                                 \
+            wxFAIL_COND_MSG(#cond, wxSysErrorMsgStr(cond));               \
+        }                                                                 \
+    wxSTATEMENT_MACRO_END
+
+#else
+
+#define wxVERIFY_NOERR(cond)                                              \
+    wxSTATEMENT_MACRO_BEGIN                                               \
+        cond;                                                             \
+    wxSTATEMENT_MACRO_END
+
+#endif
```

Comment:
I don't think we need this part at all, actually, `wxFAIL_COND_MSG` already expands to nothing when debug checks are disabled, so I believe we should just always use the definition in `#if`. Am I missing something here?

Reply:
You are absolutely right, it was a left over from previous attempts, where I used my own macros, but now it's not needed anymore